middleware/file/source/manager/admin/cli.cpp

```diff
@@ -0,0 +1,174 @@
+//! 
+//! Copyright (c) 2015, The casual project
+//!
+//! This software is licensed under the MIT license, https://opensource.org/licenses/MIT
+//!
+
+#include "file/manager/admin/cli.h"
+
+#include "file/manager/admin/model.h"
+#include "file/manager/admin/services.h"
+
+#include "common/log/line.h"
+
+#include "common/terminal.h"
+
+#include "serviceframework/service/protocol/call.h"
+
+#include <iostream>
+#include <ranges>
+
+namespace casual
+{
+   namespace file::manager::admin::cli
+   {
+      namespace local
+      {
+         namespace
+         {
+            namespace detail
+            {
+               namespace call
+               {
+                  using Call = serviceframework::service::protocol::binary::Call;
+
+                  manager::admin::model::State state()
+                  {
+                     return Call{}( 
+                        manager::admin::service::name::state).extract< manager::admin::model::State>();
+                  }
+
+                  std::vector< common::transaction::global::ID> recover( 
+                     std::vector< common::transaction::global::ID> gtrids,
+                     manager::admin::model::recovery::Directive directive)
+                  {
+                     return Call{}( 
+                        manager::admin::service::name::recover,
+                        std::move( gtrids),
+                        std::move( directive)).extract< std::vector< common::transaction::global::ID>>();
+                  }
+               } // call
+
+               namespace format
+               {
+                  auto requests( const manager::admin::model::State& state)
+                  {
+                     const auto color = common::terminal::output::directive().porcelain() ? 
+                        common::terminal::color::yellow : 
+                        common::terminal::color::no_color;
+
```

Comment:
You don't need to check porcelain at all. Use the colors you want, the colors will be ignored if user uses `--porcelain`

In a few places we check for _porcelain_ to keep backward compatibility -> the order of stuff is the same as before when using _porcelain_. You don't have to bother with this since it's all new stuff.